LR-5948 Upgrades phpunit to version 7.5

dk-jessn opened the pull request:
> Why is this change proposed? How will it affect the code base? Which results are expected from the change? Include context, UI before/after screenshots, assumptions, knowledge, research, links, references, etc.

Upgrades phpunit to version 7.5

It upgrades phpunit to a version that is aligned with Laravel 5.8 respectively.

Commit:
Upgrades phpunit to version 7.5

It upgrades phpunit to a version that is aligned with
Laravel 5.8 respectively.

Commit:
Updates travis til php 7.1

Commit:
Updates travis to php 7.1

Commit:
Upgrades to php7.3 as phpunit7.5 only supports 7.2 and 7.3

Commit:
Updates travis to php7.3

Commit:
Makes unittests compatible with php7.3 and unittest7.5

Commit:
Fixes unittests compatibility with newer phpunit version

Commit:
adds the void return type to setUp() respectively

Commit:
Makes tests comptible with Php7.3 and Phpunit7.5

clauswitt reviewed (changes requested):
There is a warning in the test run (for libraries it could be an idea to make phpunit fail on warnings). 

This may become a potential issue later - and is easy to fix with such a small test suite.

https://travis-ci.org/ArnsboMedia/opengraph_parser/builds/623611915#L223

jni-heyloyalty commented:
@clauswitt Nice catch, I will change it to let it fail on warnings

Commit:
Sets stopOnFailure to true and change an annotation

Instead of using annotation @expectException, it does now
call expectException() hence the annotation is deprecated.

Commit:
Merge branch '20191211-phpunit75upg' of github.com:ArnsboMedia/opengraph_parser into 20191211-phpunit75upg

Commit:
Bumping PHP version down to 7.1 for compatibility

Commit:
Revers back op php 7.3 as codeclimate/php-test-reporter package does not support php 7.1
